bot/cogs/candy_collection.py

```diff
@@ -0,0 +1,185 @@
+import discord
+from discord.ext import commands
+import random
+import json
+import functools
+import os
+
+json_location = os.path.join(os.getcwd(),'resources', 'candy_collection.json')
+
+
+class CandyCollection:
+    def __init__(self, bot):
+        self.bot = bot
+        with open(json_location) as candy:
+            self.candy_json = json.load(candy)
+            self.msg_reacted = self.candy_json['msg_reacted']
+        self.get_candyinfo = dict()
+        for userinfo in self.candy_json['records']:
+            userid = userinfo['userid']
+            self.get_candyinfo[userid] = userinfo
+
+    HACKTOBER_CHANNEL_ID = 496432022961520650
+    # chance is 1 in x range, so 1 in 20 range would give 5% chance (for add candy)
+    ADD_CANDY_REACTION_CHANCE = 20  # 5%
+    ADD_CANDY_EXISTING_REACTION_CHANCE = 10  # 10%
+    ADD_SKULL_REACTION_CHANCE = 50  # 2%
+    ADD_SKULL_EXISTING_REACTION_CHANCE = 20  # 5%
+
+    async def on_message(self, message):
+        """Make sure the user is not a bot and
+            the channel is #event-hacktoberfest.
+            Because the skull has a lower chance of occurring
+            we'll check for that first, and then add respective reactions"""
```

Comment:
the first line of the docstring should describe what the event handler actually _does_. See PEP257 for guidelines. In simple terms, it should say something like "Randomly adds a candy or a skull to some new messages."

The stuff about making sure the user isn't a bot should be a block comment above the relevant bit of code.